core/geometry/src/curve/internalContexts/EllipticalArcApproximationContext.ts

```diff
@@ -0,0 +1,1106 @@
+    this._localToWorld = Transform.createRefs(data.center, data.axes);
+    if (this._localToWorld.matrix.isSingular())
+      return;
+    if (this._arc.sweep.isEmpty)
+      return; // ellipse must have a nonzero sweep
+    const xMag2 = arc.matrixRef.columnXMagnitudeSquared();
+    const yMag2 = arc.matrixRef.columnYMagnitudeSquared();
+    if (Geometry.isSmallMetricDistanceSquared(xMag2) || Geometry.isSmallMetricDistanceSquared(yMag2))
+      return; // ellipse must have positive radii
+    if (Geometry.isSameCoordinateSquared(xMag2, yMag2))
+      return; // ellipse must not be circular
+    this._isValidArc = true;
+  }
+  /** Constructor, clones input. */
+  public static create(arc: Arc3d) {
+    return new EllipticalArcApproximationContext(arc);
+  }
+  /** The arc to be sampled. Its axes are forced to be perpendicular. */
+  public get arc(): Arc3d {
+    return this._arc;
+  }
+  /** The rigid transformation that rotates and translates the arc into the xy-plane at the origin. */
+  public get localToWorld(): Transform {
+    return this._localToWorld;
+  }
+  /** Whether the arc is amenable to sampling. */
+  public get isValidArc(): boolean {
+    return this._isValidArc;
+  }
+  /**
+   * Create a clone of the valid arc in local coordinates.
+   * @param fullSweep Optionally set full sweep on the returned local arc.
+   */
+  public cloneLocalArc(fullSweep?: boolean): Arc3d {
+    const worldToLocal = this.localToWorld.inverse()!;
+    const arcXY = this.arc.cloneTransformed(worldToLocal);
+    if (fullSweep)
+      arcXY.sweep.setStartEndRadians();
+    return arcXY;
+  }
+  /** Process structured sample data for the given elliptical arc. */
+  public static processQuadrantFractions(
+    ellipticalArc: Arc3d, samples: QuadrantFractions[], processor: QuadrantFractionsProcessor,
+  ): void {
+    const pt0 = this.workPt0;
+    const pt1 = this.workPt1;
+    const pt2 = this.workPt2;
+    const ray = this.workRay;
+    const arcBetween2Samples = (arcStart: Ray3d, arcEnd: Point3d, reverse: boolean): Arc3d => {
+      // assume non-colinear inputs
+      const myArc = Arc3d.createCircularStartTangentEnd(arcStart.origin, arcStart.direction, arcEnd) as Arc3d;
+      if (reverse)
+        myArc.reverseInPlace();
+      return myArc;
+    };
+    const createFirstArc = (f0: number, f1: number, reverse: boolean) => {
+      ellipticalArc.fractionToPointAndDerivative(f0, ray);
+      pt0.setFrom(ray.origin);
+      pt2.setFrom(ellipticalArc.fractionToPoint(f1, pt1)); // set pt2 in case no inner arcs
+      if (processor.announceArc) {
+        if (reverse)
+          ray.direction.scaleInPlace(-1);
+        processor.announceArc(arcBetween2Samples(ray, pt1, false), f0, f1);
+      }
+      if (processor.announceChord)
+        processor.announceChord(pt0, pt1, f0, f1);
+      return { p0: pt0, p1: pt1, p2: pt2 };
+    };
+    const arcBetweenLast2Of3Samples = (firstPt: Point3d, arcStart: Point3d, arcEnd: Point3d): Arc3d => {
```

Comment:
I found this name confusing. I know we use 3 points to create the partial circle and then only take part of it from middle to end. However, since we use 2 points for first and last partial circle, this name confuses me.

Also in general the word `arc` is used for both original arc that we are approximating and also for the partial circles and that's confusing too!

Reply:
I'm currently taking suggestions for better distinction of "arcBetween2Samples" and "arcBetweenLast2Of3Samples"!

Yeah, if you find specific doc that needs "arc" clarification, I can fix.  At some point I got tired of writing "elliptical arc" and "circular arc" and assumed reader would know which is inferred from the context.